### DISCUSSION TITLE
X-Accel Support in NPM for ProjectSend?

### DISCUSSION THREAD

**Discussion by shanelord01** [NginxProxyManager/nginx-proxy-manager#2918] [Q&A] [answered]
Hi everyone,

I'm struggling to get X-Accel support working for ProjectSend. PHP file serving is working currently, but I want to get a more optimised outcome as per here: https://docs.projectsend.org/about/installation/advanced-improving-downloads

When I go to enable it in the ProjectSend, it tells me to add this code to the server block (ie NGINX config):

```
location /serve-file {
    internal;
    alias /app/www/public/upload/;
}
```

I've tried adding this to a custom location both with a "/" location (with the above input in the code box) as well as a "/serve-file" location with the forward hostname/IP set as "192.168.1.101/app/www/public/upload/"

Neither way is working - in both instances the server supplies a 6KB file instead of the actual file to be downloaded.

I'm sure I'm being stupid, but any guidance anyone can give?

Thanks.

**Comment by maxcynicle** [comment, marked as answer]
Hi @shanelord01 , I was struggling with Nginx Proxy Manager, ProjectSend, and X-ACCEL for like 8 hours as well, until it hit me that I was supposed to add the code to the nginx .conf of the webserver running ProjectSend itself, not the NPM. I deployed ProjectSend using a docker hub image, so I quickly got onto the container's console and edited /config/nginx/site-confs/default.conf to add the code. After restarting the container, it finally works! It doesn't return the 6KB index.php anymore, and the correct file gets downloaded. I'm including the file below just in case

```
root@64ca355195e0:/# cat /config/nginx/site-confs/default.conf
## Version 2023/04/13 - Changelog: https://github.com/linuxserver/docker-baseimage-alpine-nginx/commits/master/root/defaults/nginx/site-confs/default.conf.sample

server {
    listen 80 default_server;
    listen [::]:80 default_server;

    listen 443 ssl http2 default_server;
    listen [::]:443 ssl http2 default_server;

    server_name _;

    include /config/nginx/ssl.conf;

    set $root /app/www/public;
    if (!-d /app/www/public) {
        set $root /config/www;
    }
    root $root;
    index index.html index.htm index.php;

    location / {
        # enable for basic auth
        #auth_basic "Restricted";
        #auth_basic_user_file /config/nginx/.htpasswd;

        try_files $uri $uri/ /index.html /index.php$is_args$args =404;
    }

    location ~ ^(.+\.php)(.*)$ {
        fastcgi_split_path_info ^(.+\.php)(.*)$;
        fastcgi_pass 127.0.0.1:9000;
        fastcgi_index index.php;
        include /etc/nginx/fastcgi_params;
    }

    location /serve-file {
        internal;
        alias /app/www/public/upload/;
}

    # deny access to .htaccess/.htpasswd files
    location ~ /\.ht {
        deny all;
    }
}
root@64ca355195e0:/# 
```

No changes were made to the proxy host in NPM.

**Reply by shanelord01** [reply]
You are an absolute legend! Thanks!
